cob_hardware_emulation/scripts/emulation_move_base.py

```diff
@@ -0,0 +1,122 @@
+        increment = 1
+        if increments > 0:
+            while not rospy.is_shutdown():
+                new_pose.position.x = self.odom.pose.pose.position.x + d_x / increments
+                new_pose.position.y = self.odom.pose.pose.position.y + d_y / increments
+                yaw = tf.transformations.euler_from_quaternion([self.odom.pose.pose.orientation.x, self.odom.pose.pose.orientation.y, self.odom.pose.pose.orientation.z, self.odom.pose.pose.orientation.w])[2] + d_yaw / increments
+                quat = tf.transformations.quaternion_from_euler(0, 0, yaw)
+                new_pose.orientation.x = quat[0]
+                new_pose.orientation.y = quat[1]
+                new_pose.orientation.z = quat[2]
+                new_pose.orientation.w = quat[3]
+                rospy.logdebug("increment %d of %d with new pose: %f %f %f"%(increment, increments, new_pose.position.x, new_pose.position.y, yaw))
+                self.odom.pose.pose = new_pose
+                # we're moving, so we set a non-zero twist
+                self.odom.twist.twist.linear.x = d_x / movement_time
+                self.odom.twist.twist.linear.y = d_y / movement_time
+                self.odom.twist.twist.angular.z = d_yaw / movement_time
+
+                if increment >= increments:
+                    break
+                increment += 1
+                r.sleep()
+        else:
+            self.odom.pose.pose = new_pose
+
+        # reset twist as we're not moving anymore
+        self.odom.twist.twist = Twist()
+
+        # check if we reached the target goal
+        #if self.odom == goal:
+        #    self.as_move_base.set_succeeded(MoveBaseActionResult())
+        #else:
+        #    self.as_move_base.set_aborted(MoveBaseActionResult())
+        rospy.loginfo("goal reached in %f seconds", movement_time)
+        self.as_move_base.set_succeeded(MoveBaseActionResult())
+
+    def timer_cb(self, event):
+        # publish odometry
+        msg = copy.deepcopy(self.odom)
+        msg.header.stamp = rospy.Time.now() # update to current time stamp
+        self.pub_odom.publish(msg)
+        
+        # publish tf
+        # pub base_footprint --> odom_combined
+        self.br.sendTransform((self.odom.pose.pose.position.x, self.odom.pose.pose.position.y, 0),
+                     (self.odom.pose.pose.orientation.x, self.odom.pose.pose.orientation.y, self.odom.pose.pose.orientation.z, self.odom.pose.pose.orientation.w),
+                     rospy.Time.now(),
+                     "base_footprint",
+                     "odom_combined")
+        
+        # pub odom_combined --> map
+        # we emulate 'perfect' odometry, so /odom_combined is always the same as /map
+        self.br.sendTransform((0, 0, 0),
+                     tf.transformations.quaternion_from_euler(0, 0, 0),
+                     rospy.Time.now(),
+                     "odom_combined",
+                     "map")
+
+if __name__ == '__main__':
+    rospy.init_node('emulation')
```

Comment:
use specific node names for both emulators, e.g. `move_base_emulation` (and `follow_joint_trajectory_emulation` respectively)